articles/types/jsdoc.md

```diff
@@ -1,35 +1,37 @@
 # JsDoc support
 
 Programs written in JavaScript and TypeScript can be annotated with (JsDoc comments)[http://usejsdoc.org/index.html]
-to give type information and quick-info descriptions of identifiers.
+to give type information and quickinfo descriptions of identifiers.
 
-In the following example, the text between the opening `/**` and closing `/*` are considered JsDoc comments.
+In the following example, the text between the opening `/**` and closing `*/` are considered JsDoc comments.
 
 ```js
 /**
- * @param {"cat" | "dog"} kind  The type of animal to create.
- * @param name {string}
- * @param {number} weight The weight, in pounds, of animal.
- * @param {...string} extraProps Any extra properties to tack on the animal.
- * @return {{[index: string]: string | number}} Any extra properties to tack on the animal.
+ * Creates a new animal object.
+ * @param {string} name The name of the new animal.
+ * @param {number} weight
+ * @param {"cat" | "dog"} kind
+ * @return {{name: string, weight: number, kind: "cat" | "dog"}} A brand new animal!
  */
-function createAnimal(kind, name, weight, ...extraProps)
+function createAnimal(name, weight, kind)
 {
     // ...
 }
 ```
 
 As you can see, we can prefix sections in JsDoc with terms such as `@param` and `@return` to describe the behavior of the `createAnimal` function.
 In JsDoc, terms beginning with `@` have special meanings, and are called *tags*. The text between curly braces are descriptions of *types*,
-which can be used by the developer to check that they are using a variable correctly -- they can go either before or after the construct's *name*.
+which can be used by the developer to check that they are using a variable correctly &mdash;  they should precede the *name* of the construct being described, if applicable.
 In the example above, knowing that `weight` is a number can inform the developer to avoid calling `weight.length`.
-Moreover, if the result of `createAnimal` is assigned to a variable, the return type of createAnimal propogates to the assigned value.
+Moreover, if the result of `createAnimal` is assigned to a variable, the return type of createAnimal propagates to the assigned value.
 
-Finally, the text after the parameter names are *descriptions*, which show up as part of *quickinfo* anytime you hover over a construct that is documented
+The type annotations supported include both jsdoc and TypeScript syntax. For example, `kind` is described as a union of string-literals, and the return type is an object containing each of the parameters as properties. We give details in (section link?) below.
```

Comment:
1. capitalise jsdoc correctly
2. string-literals doesn't usually have a dash
3. the object literal type is standard, and string literals are an easier example to grasp of Typescript's superior types. I'd just mention `kind` here.